File: code/459.重复的子字符串.cpp
/*
 * @lc app=leetcode.cn id=459 lang=cpp
 *
 * [459] 重复的子字符串
 */
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        int len = s.size();
        for(int t = 1; t <= len/2; ++t)
        {
            if(len % t)
                continue;
            int i;
            for(i = t; i < len && s[i] == s[i%t]; ++i) ;
            if(i == len)
                return true;
        }
        return false;
    }
};

File: code/459.重复的子字符串.py
#
# @lc app=leetcode.cn id=459 lang=python
#
# [459] 重复的子字符串
#
class Solution(object):
    def repeatedSubstringPattern(self, s):
        """
        :type s: str
        :rtype: bool
        """
        l = len(s)
        for t in range(1, l/2+1):
            if l % t != 0:
                continue
            for i in range(t, l):
                if s[i] == s[i%t]:
                    i += 1
                else:
                    break
            if i == l:
                return True
        return False
        

